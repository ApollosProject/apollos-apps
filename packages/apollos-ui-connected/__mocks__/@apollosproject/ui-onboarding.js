export const checkOnboardingStatusAndNavigate = () => null;
export const onboardingComplete = () => null;
