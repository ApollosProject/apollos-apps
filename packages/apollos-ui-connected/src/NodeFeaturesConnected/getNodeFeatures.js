import gql from 'graphql-tag';
import ApollosConfig from '@apollosproject/config';

const {
  TEXT_FEATURE_FRAGMENT,
  SCRIPTURE_FEATURE_FRAGMENT,
  WEBVIEW_FEATURE_FRAGMENT,
  FEATURES_FRAGMENT,
  NODE_FEATURES_FRAGMENT,
} = ApollosConfig.FRAGMENTS;

export default gql`
  query nodeFeatures($nodeId: ID!) {
    node(id: $nodeId) {
      id
      ...NodeFeaturesFragment
    }
  }
  ${TEXT_FEATURE_FRAGMENT}
  ${SCRIPTURE_FEATURE_FRAGMENT}
  ${WEBVIEW_FEATURE_FRAGMENT}
  ${FEATURES_FRAGMENT}
  ${NODE_FEATURES_FRAGMENT}
`;
