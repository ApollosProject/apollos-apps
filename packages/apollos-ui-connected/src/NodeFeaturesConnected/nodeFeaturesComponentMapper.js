import React from 'react';
import PropTypes from 'prop-types';

import TextFeature from '../features/TextFeature';
import ScriptureFeature from '../features/ScriptureFeature';
import WebviewFeature from '../features/WebviewFeature';

const MAPPINGS = {
  TextFeature,
  ScriptureFeature,
  WebviewFeature,
};

const featuresFeedComponentMapper = ({
  feature,
  onPressActionItem,
  additionalFeatures = {},
}) => {
  const { id, __typename, ...featureData } = feature;
  const map = { ...MAPPINGS, ...additionalFeatures };
  const Component = map[__typename];

  if (Component) {
    return (
      <Component
        featureId={id}
        onPressItem={onPressActionItem}
        listKey={id}
        nodeId={id}
        {...featureData}
      />
    );
  }

  console.warn(
    `No feature found for ${__typename}. Do you need to pass it in via the additionalFeatures prop?`
  );
  return null;
};

featuresFeedComponentMapper.propTypes = {
  feature: PropTypes.shape({
    userFeedFeatures: PropTypes.arrayOf(PropTypes.shape({})),
  }).isRequired,
  onPressActionItem: PropTypes.func,
  loading: PropTypes.bool,
  additionalFeatures: PropTypes.shape({}),
};

export default featuresFeedComponentMapper;
