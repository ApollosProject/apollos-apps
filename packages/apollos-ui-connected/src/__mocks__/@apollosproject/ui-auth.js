export const ProtectedTouchable = ({ children }) => children;
