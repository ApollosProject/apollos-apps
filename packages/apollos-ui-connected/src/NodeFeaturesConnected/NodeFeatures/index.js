import React, { memo } from 'react';
import PropTypes from 'prop-types';
import { H3, PaddedView, withIsLoading } from '@apollosproject/ui-kit';
import nodeFeaturesComponentMapper from '../nodeFeaturesComponentMapper';

const ContentSingleFeatures = memo(
  ({ features, title, additionalFeatures, nodeId }) => (
    <PaddedView horizontal={false}>
      {title && (
        <PaddedView vertical={false}>
          <H3 padded>{title}</H3>
        </PaddedView>
      )}
      {features.map((feature) =>
        nodeFeaturesComponentMapper({
          feature,
          additionalFeatures,
          nodeId,
        })
      )}
    </PaddedView>
  )
);

ContentSingleFeatures.propTypes = {
  features: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.string.isRequired,
      order: PropTypes.string,
      __typename: PropTypes.string.isRequired,
    })
  ),
  featureMap: PropTypes.shape({}),
  additionalFeatures: PropTypes.shape({}),
  nodeId: PropTypes.string,
  title: PropTypes.string,
};

ContentSingleFeatures.defaultProps = {
  title: 'Engage',
  additionalFeatures: {},
};

ContentSingleFeatures.displayName = 'ContentSingleFeatures';

export default withIsLoading(ContentSingleFeatures);
