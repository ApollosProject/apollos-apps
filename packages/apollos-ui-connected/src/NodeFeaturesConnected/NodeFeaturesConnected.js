import React from 'react';
import PropTypes from 'prop-types';
import { Query } from 'react-apollo';
import { ErrorCard } from '@apollosproject/ui-kit';
import { get } from 'lodash';

import GET_NODE_FEATURES from './getNodeFeatures';
import NodeFeatures from './NodeFeatures';

const NodeFeaturesConnected = ({ Component, nodeId, ...props }) => {
  if (!nodeId) return null;

  return (
    <Query
      query={GET_NODE_FEATURES}
      fetchPolicy="cache-and-network"
      variables={{ nodeId }}
    >
      {({ data: { node } = {}, loading, error }) => {
        if (error) return <ErrorCard error={error} />;
        // TODO: set an optimistic response for this query to return a visually appeallying empty query so we can enable loading states
        if (loading) return null;

        const features = get(node, 'features', []);
        if (!features || !features.length) return null;

        return (
          console.warn(
            'NodeFeaturesConnected is deprecated. Please use FeaturesFeedConnected.'
          ) || <Component nodeId={nodeId} features={features} {...props} />
        );
      }}
    </Query>
  );
};

NodeFeaturesConnected.propTypes = {
  Component: PropTypes.oneOfType([PropTypes.node, PropTypes.func]),
  nodeId: PropTypes.string,
};

NodeFeaturesConnected.defaultProps = {
  Component: NodeFeatures,
};

export default NodeFeaturesConnected;
