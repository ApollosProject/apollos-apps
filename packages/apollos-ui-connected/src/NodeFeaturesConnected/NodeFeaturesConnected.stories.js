import React from 'react';
import { storiesOf } from '@apollosproject/ui-storybook';

import { BackgroundView, CenteredView } from '@apollosproject/ui-kit';
import { ApolloStorybookDecorator } from '../testUtils';
import NodeFeaturesConnected from './NodeFeaturesConnected';

const mocks = {
  Query: () => ({
    node: (root, args) => ({
      id: args.id,
      __typename: 'WeekendContentItem',
      features: [
        {
          id: 'TextFeature:123',
          __typename: 'TextFeature',
          body:
            'Quibusdam nemo quidem sed veritatis itaque. Dolorum quia numquam ea doloremque nostrum numquam non eveniet. Fuga qui voluptatibus fugiat blanditiis ut.',
        },
        {
          id: 'ScriptureFeature:123',
          __typename: 'ScriptureFeature',
          scriptures: [
            {
              id: 'Scripture:123',
              reference: 'Psalm 86:1',
              version: 'ESV',
              html: `<div class="scripture-styles">
  <p class="s">A Prayer for Help</p><p class="q1">Listen to me, <span class="nd">Lord</span>, and answer me,</p><p class="q2">for I am helpless and weak.</p>
</div>`,
            },
          ],
        },
        {
          id: 'WebviewFeature:123',
          __typename: 'WebviewFeature',
          linkText: 'Check this out',
          url: 'https://apollos.app/',
          title: 'Our Homepage',
        },
      ],
    }),
  }),
};

storiesOf('ui-connected/NodeFeaturesConnected', module)
  .addDecorator((story) => (
    <BackgroundView>
      {/* eslint-disable-next-line react-native/no-inline-styles */}
      <CenteredView style={{ alignItems: 'stretch' }}>{story()}</CenteredView>
    </BackgroundView>
  ))
  .addDecorator(ApolloStorybookDecorator({ mocks }))
  .add('example', () => (
    <NodeFeaturesConnected nodeId={'UniversalContentItem:123'} />
  ))
  .add('no title', () => (
    <NodeFeaturesConnected title={null} nodeId={'UniversalContentItem:123'} />
  ));
