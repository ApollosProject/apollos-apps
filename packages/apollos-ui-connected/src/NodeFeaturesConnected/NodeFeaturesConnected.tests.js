import React from 'react';
import { Providers, renderWithApolloData } from '@apollosproject/ui-test-utils';
import { MockedProvider } from '@apollo/client/testing';

import GET_NODE_FEED from './getNodeFeatures';

import NodeFeaturesConnected from './NodeFeaturesConnected';

jest.mock('../FeaturesFeedConnected', () => 'FeaturesFeedConnected');

describe('NodeFeaturesConnected', () => {
  it('should render', async () => {
    const nodeMock = {
      request: {
        query: GET_NODE_FEED,
        variables: {
          nodeId: 'WeekendContentItem:123',
        },
      },
      result: {
        data: {
          node: {
            id: 'WeekendContentItem:123',
            __typename: 'WeekendContentItem',
            featureFeed: {
              __typename: 'FeatureFeed',
              id: 'FeatureFeed:123',
            },
          },
        },
      },
    };
    const tree = await renderWithApolloData(
      <Providers mocks={[nodeMock]} MockedProvider={MockedProvider}>
        <NodeFeaturesConnected nodeId={'WeekendContentItem:123'} />
      </Providers>
    );
    expect(tree).toMatchSnapshot();
  });
});
