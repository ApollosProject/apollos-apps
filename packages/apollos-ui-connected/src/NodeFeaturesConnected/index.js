import NodeFeaturesConnected from './NodeFeaturesConnected';

export NodeFeatures from './NodeFeatures';
export GET_NODE_FEATURES from './getNodeFeatures';

export default NodeFeaturesConnected;
