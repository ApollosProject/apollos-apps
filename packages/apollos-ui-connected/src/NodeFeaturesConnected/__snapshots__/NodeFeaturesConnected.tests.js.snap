// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NodeFeaturesConnected should render 1`] = `
<RCTScrollView
  data={
    Array [
      Object {
        "__typename": "WebviewFeature",
        "id": "WebviewFeature:6",
        "linkText": "Text Link Text",
        "title": "Test Title",
        "url": "https://open.spotify.com/embed/user/spotify/playlist/37i9dQZF1DWWvHBEQLnV1N",
      },
      Object {
        "__typename": "TextFeature",
        "body": "this is another, text feature",
        "id": "TextFeature:1",
        "sharing": Object {
          "__typename": "SharableFeature",
          "message": "this is another, text feature",
        },
        "title": "title",
      },
      Object {
        "__typename": "ScriptureFeature",
        "id": "ScriptureFeature:3",
        "scriptures": Array [
          Object {
            "__typename": "Scripture",
            "copyright": "PUBLIC DOMAIN except in the United Kingdom, where a Crown Copyright applies to printing the KJV. See http://www.cambridge.org/about-us/who-we-are/queens-printers-patent",
            "html": "<p class=\\"p\\"><span data-number=\\"16\\" class=\\"v\\">16</span><span class=\\"wj\\">Â¶ For God so loved the world, that he gave his only begotten Son, that whosoever believeth in him should not perish, but have everlasting life.</span> </p>",
            "id": "Scripture:4",
            "reference": "Mark 1:12",
            "version": "KJV",
          },
        ],
        "sharing": Object {
          "__typename": "SharableFeature",
          "message": "16For God so loved the world, that he gave his only born Son, that whoever believes in him should not perish, but have eternal life.   Luke 1:2",
        },
        "title": "title",
      },
      Object {
        "__typename": "ScriptureFeature",
        "id": "ScriptureFeature:4",
        "scriptures": Array [
          Object {
            "__typename": "Scripture",
            "copyright": "PUBLIC DOMAIN",
            "html": "<p class=\\"p\\"><span data-number=\\"16\\" class=\\"v\\">16</span><span class=\\"wj\\">For God so loved the world, that he gave his only born</span> <span class=\\"wj\\">Son, that whoever believes in him should not perish, but have eternal life, forever. </span> </p>",
            "id": "Scripture:5",
            "reference": "John 3:16",
            "version": "WEB",
          },
        ],
        "sharing": Object {
          "__typename": "SharableFeature",
          "message": "16For God so loved the world, that he gave his only born Son, that whoever believes in him should not perish, but have eternal life.   John 3:16",
        },
        "title": "title",
      },
    ]
  }
  disableVirtualization={false}
  getItem={[Function]}
  getItemCount={[Function]}
  horizontal={false}
  initialNumToRender={10}
  keyExtractor={[Function]}
  loading={false}
  maxToRenderPerBatch={10}
  onContentSizeChange={[Function]}
  onEndReachedThreshold={0.7}
  onLayout={[Function]}
  onMomentumScrollEnd={[Function]}
  onRefresh={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  refreshControl={
    <RefreshControlMock
      onRefresh={[Function]}
      refreshing={false}
    />
  }
  refreshing={false}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={Array []}
  updateCellsBatchingPeriod={50}
  viewabilityConfigCallbackPairs={Array []}
  windowSize={21}
>
  <RCTRefreshControl />
  <View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        height={400}
        style={
          Object {
            "backgroundColor": "#000000",
            "borderRadius": 16,
            "height": 400,
            "marginHorizontal": 16,
            "marginVertical": 12,
            "overflow": "hidden",
            "shadowColor": "rgba(0, 0, 0, 0.09999999999999998)",
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 1,
            "shadowRadius": 8,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "flexWrap": "nowrap",
              "justifyContent": "space-between",
            }
          }
        >
          <Text
            padded={true}
            style={
              Object {
                "color": "#F8F7F4",
                "fontFamily": "InterUI-Black",
                "fontSize": 24,
                "lineHeight": 27.6,
                "paddingBottom": 18,
                "paddingHorizontal": 16,
                "paddingTop": 27,
              }
            }
          >
            Test Title
          </Text>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#17B582",
                  "fontFamily": "System",
                  "fontSize": 14,
                  "fontStyle": null,
                  "fontWeight": null,
                  "lineHeight": 21,
                  "paddingHorizontal": 16,
                }
              }
            >
              Text Link Text
            </Text>
          </View>
        </View>
        <View
          style={
            Array [
              Object {
                "flex": 1,
                "overflow": "hidden",
              },
              undefined,
            ]
          }
        >
          <RNCWebView
            cacheEnabled={true}
            javaScriptEnabled={true}
            messagingEnabled={false}
            onContentProcessDidTerminate={[Function]}
            onHttpError={[Function]}
            onLoadingError={[Function]}
            onLoadingFinish={[Function]}
            onLoadingProgress={[Function]}
            onLoadingStart={[Function]}
            onMessage={[Function]}
            onShouldStartLoadWithRequest={[Function]}
            source={
              Object {
                "uri": "https://open.spotify.com/embed/user/spotify/playlist/37i9dQZF1DWWvHBEQLnV1N",
              }
            }
            style={
              Array [
                Object {
                  "flex": 1,
                  "overflow": "hidden",
                },
                Object {
                  "backgroundColor": "#ffffff",
                },
                Object {
                  "flex": 1,
                },
              ]
            }
            useSharedProcessPool={true}
          />
        </View>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          Object {
            "backgroundColor": "#FFFFFF",
            "borderRadius": 16,
            "marginHorizontal": 16,
            "marginVertical": 12,
            "shadowColor": "rgba(0, 0, 0, 0.09999999999999998)",
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 1,
            "shadowRadius": 8,
          }
        }
      >
        <View
          style={
            Object {
              "borderRadius": 16,
              "overflow": "hidden",
            }
          }
        >
          <View
            style={
              Object {
                "paddingHorizontal": 16,
                "paddingVertical": 16,
              }
            }
          >
            <View
              flexed={false}
              style={
                Object {
                  "alignItems": "center",
                  "flex": null,
                  "flexDirection": "row",
                  "paddingBottom": 16,
                }
              }
            >
              <View
                style={
                  Object {
                    "paddingRight": 4,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "#00676D",
                      "fontFamily": "InterUI-Bold",
                      "fontSize": 12,
                      "lineHeight": 18,
                    }
                  }
                  tint="#00676D"
                >
                  title
                </Text>
              </View>
            </View>
            <Text
              bold={false}
              italic={false}
              style={
                Object {
                  "color": "#303030",
                  "fontFamily": "InterUI-Regular",
                  "fontSize": 16,
                  "lineHeight": 26,
                }
              }
            >
              this is another, text feature
            </Text>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "flexWrap": "wrap",
                "justifyContent": "flex-end",
                "paddingHorizontal": 16,
                "paddingTop": 0,
                "paddingVertical": 16,
              }
            }
          />
        </View>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          Object {
            "backgroundColor": "#FFFFFF",
            "borderRadius": 16,
            "marginHorizontal": 16,
            "marginVertical": 12,
            "shadowColor": "rgba(0, 0, 0, 0.09999999999999998)",
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 1,
            "shadowRadius": 8,
          }
        }
      >
        <View
          style={
            Object {
              "borderRadius": 16,
              "overflow": "hidden",
            }
          }
        >
          <View
            style={
              Object {
                "paddingHorizontal": 16,
                "paddingVertical": 16,
              }
            }
          >
            <View
              flexed={false}
              style={
                Object {
                  "alignItems": "center",
                  "flex": null,
                  "flexDirection": "row",
                  "paddingBottom": 16,
                }
              }
            >
              <RNSVGSvgView
                align="xMidYMid"
                bbHeight={19.2}
                bbWidth={19.2}
                height={19.2}
                meetOrSlice={0}
                minX={0}
                minY={0}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "borderWidth": 0,
                    },
                    undefined,
                    null,
                    Object {
                      "flex": 0,
                      "height": 19,
                      "width": 19,
                    },
                  ]
                }
                vbHeight={24}
                vbWidth={24}
                width={19.2}
              >
                <RNSVGGroup
                  fill={
                    Array [
                      0,
                      4278190080,
                    ]
                  }
                  fillOpacity={1}
                  fillRule={1}
                  font={Object {}}
                  matrix={
                    Array [
                      1,
                      0,
                      0,
                      1,
                      0,
                      0,
                    ]
                  }
                  opacity={1}
                  originX={0}
                  originY={0}
                  propList={Array []}
                  rotation={0}
                  scaleX={1}
                  scaleY={1}
                  skewX={0}
                  skewY={0}
                  stroke={null}
                  strokeDasharray={null}
                  strokeDashoffset={null}
                  strokeLinecap={0}
                  strokeLinejoin={0}
                  strokeMiterlimit={4}
                  strokeOpacity={1}
                  strokeWidth={1}
                  vectorEffect={0}
                  x={0}
                  y={0}
                >
                  <RNSVGPath
                    d="M4.6 3.64v16.356c0 .004 0 .004-.003.004h14.806c-.004 0-.003 0-.003-.004V3.64c0-.004 0-.004.003-.004H4.597c.004 0 .003 0 .003.004zm-1.6 0C3 2.734 3.717 2 4.597 2h14.806C20.285 2 21 2.734 21 3.64v16.356c0 .906-.717 1.64-1.597 1.64H4.597c-.882 0-1.597-.734-1.597-1.64V3.64zm4 3.27c0 .45.366.817.818.817h8.364c.452 0 .818-.366.818-.818 0-.454-.366-.82-.818-.82H7.818c-.452 0-.818.366-.818.82zm0 4.908c0 .452.366.818.818.818h8.364c.452 0 .818-.366.818-.818 0-.452-.366-.818-.818-.818H7.818c-.452 0-.818.366-.818.818zm0 4.91c0 .45.366.817.818.817h4.764c.452 0 .818-.366.818-.818 0-.452-.366-.818-.818-.818H7.818c-.452 0-.818.364-.818.816z"
                    fill={
                      Array [
                        0,
                        4278216557,
                      ]
                    }
                    fillOpacity={1}
                    fillRule={1}
                    matrix={
                      Array [
                        1,
                        0,
                        0,
                        1,
                        0,
                        0,
                      ]
                    }
                    opacity={1}
                    originX={0}
                    originY={0}
                    propList={
                      Array [
                        "fill",
                      ]
                    }
                    rotation={0}
                    scaleX={1}
                    scaleY={1}
                    skewX={0}
                    skewY={0}
                    stroke={null}
                    strokeDasharray={null}
                    strokeDashoffset={null}
                    strokeLinecap={0}
                    strokeLinejoin={0}
                    strokeMiterlimit={4}
                    strokeOpacity={1}
                    strokeWidth={1}
                    vectorEffect={0}
                    x={0}
                    y={0}
                  />
                </RNSVGGroup>
              </RNSVGSvgView>
              <View
                style={
                  Object {
                    "paddingHorizontal": 4,
                  }
                }
                withIcon="text"
              >
                <Text
                  style={
                    Object {
                      "color": "#00676D",
                      "fontFamily": "InterUI-Bold",
                      "fontSize": 12,
                      "lineHeight": 18,
                    }
                  }
                  tint="#00676D"
                >
                  title
                </Text>
              </View>
            </View>
            <View>
              <Text
                style={
                  Object {
                    "color": "#303030",
                    "fontFamily": "InterUI-Bold",
                    "fontSize": 16,
                    "lineHeight": 24,
                    "paddingBottom": 12,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "#303030",
                      "fontFamily": "InterUI-Bold",
                      "fontSize": 16,
                      "lineHeight": 24,
                    }
                  }
                >
                  Mark 1:12
                </Text>
                <Text
                  style={
                    Object {
                      "color": "#505050",
                      "fontFamily": "InterUI-Bold",
                      "fontSize": 12,
                      "lineHeight": 18,
                    }
                  }
                >
                   
                  KJV
                </Text>
              </Text>
              <View
                style={
                  Object {
                    "paddingBottom": 18,
                  }
                }
              >
                <View>
                  <View
                    style={
                      Object {
                        "paddingBottom": 0,
                      }
                    }
                  >
                    <Text
                      bold={false}
                      italic={false}
                      style={
                        Object {
                          "color": "#303030",
                          "fontFamily": "DroidSerif",
                          "fontSize": 16,
                          "lineHeight": 26,
                        }
                      }
                    >
                      <Text
                        bold={false}
                        italic={false}
                        style={
                          Object {
                            "color": "#505050",
                            "fontFamily": "DroidSerif",
                            "fontSize": 9.6,
                            "lineHeight": 26,
                          }
                        }
                      >
                         
                        <Text>
                          16
                        </Text>
                        Â 
                      </Text>
                      <Text
                        style={
                          Object {
                            "color": "#8b0000",
                          }
                        }
                      >
                        <Text>
                          Â¶ For God so loved the world, that he gave his only begotten Son, that whosoever believeth in him should not perish, but have everlasting life.
                        </Text>
                      </Text>
                      <Text
                        bold={false}
                        italic={false}
                        style={
                          Object {
                            "color": "#303030",
                            "fontFamily": "DroidSerif",
                            "fontSize": 16,
                            "lineHeight": 26,
                          }
                        }
                      >
                        <Text>
                           
                        </Text>
                      </Text>
                    </Text>
                  </View>
                </View>
              </View>
              <Text
                bold={false}
                italic={false}
                style={
                  Object {
                    "color": "#505050",
                    "fontFamily": "InterUI-Regular",
                    "fontSize": 9.6,
                    "lineHeight": 10.8,
                  }
                }
              >
                PUBLIC DOMAIN except in the United Kingdom, where a Crown Copyright applies to printing the KJV. See http://www.cambridge.org/about-us/who-we-are/queens-printers-patent
              </Text>
            </View>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "flexWrap": "wrap",
                "justifyContent": "flex-end",
                "paddingHorizontal": 16,
                "paddingTop": 0,
                "paddingVertical": 16,
              }
            }
          />
        </View>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          Object {
            "backgroundColor": "#FFFFFF",
            "borderRadius": 16,
            "marginHorizontal": 16,
            "marginVertical": 12,
            "shadowColor": "rgba(0, 0, 0, 0.09999999999999998)",
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 1,
            "shadowRadius": 8,
          }
        }
      >
        <View
          style={
            Object {
              "borderRadius": 16,
              "overflow": "hidden",
            }
          }
        >
          <View
            style={
              Object {
                "paddingHorizontal": 16,
                "paddingVertical": 16,
              }
            }
          >
            <View
              flexed={false}
              style={
                Object {
                  "alignItems": "center",
                  "flex": null,
                  "flexDirection": "row",
                  "paddingBottom": 16,
                }
              }
            >
              <RNSVGSvgView
                align="xMidYMid"
                bbHeight={19.2}
                bbWidth={19.2}
                height={19.2}
                meetOrSlice={0}
                minX={0}
                minY={0}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "borderWidth": 0,
                    },
                    undefined,
                    null,
                    Object {
                      "flex": 0,
                      "height": 19,
                      "width": 19,
                    },
                  ]
                }
                vbHeight={24}
                vbWidth={24}
                width={19.2}
              >
                <RNSVGGroup
                  fill={
                    Array [
                      0,
                      4278190080,
                    ]
                  }
                  fillOpacity={1}
                  fillRule={1}
                  font={Object {}}
                  matrix={
                    Array [
                      1,
                      0,
                      0,
                      1,
                      0,
                      0,
                    ]
                  }
                  opacity={1}
                  originX={0}
                  originY={0}
                  propList={Array []}
                  rotation={0}
                  scaleX={1}
                  scaleY={1}
                  skewX={0}
                  skewY={0}
                  stroke={null}
                  strokeDasharray={null}
                  strokeDashoffset={null}
                  strokeLinecap={0}
                  strokeLinejoin={0}
                  strokeMiterlimit={4}
                  strokeOpacity={1}
                  strokeWidth={1}
                  vectorEffect={0}
                  x={0}
                  y={0}
                >
                  <RNSVGPath
                    d="M4.6 3.64v16.356c0 .004 0 .004-.003.004h14.806c-.004 0-.003 0-.003-.004V3.64c0-.004 0-.004.003-.004H4.597c.004 0 .003 0 .003.004zm-1.6 0C3 2.734 3.717 2 4.597 2h14.806C20.285 2 21 2.734 21 3.64v16.356c0 .906-.717 1.64-1.597 1.64H4.597c-.882 0-1.597-.734-1.597-1.64V3.64zm4 3.27c0 .45.366.817.818.817h8.364c.452 0 .818-.366.818-.818 0-.454-.366-.82-.818-.82H7.818c-.452 0-.818.366-.818.82zm0 4.908c0 .452.366.818.818.818h8.364c.452 0 .818-.366.818-.818 0-.452-.366-.818-.818-.818H7.818c-.452 0-.818.366-.818.818zm0 4.91c0 .45.366.817.818.817h4.764c.452 0 .818-.366.818-.818 0-.452-.366-.818-.818-.818H7.818c-.452 0-.818.364-.818.816z"
                    fill={
                      Array [
                        0,
                        4278216557,
                      ]
                    }
                    fillOpacity={1}
                    fillRule={1}
                    matrix={
                      Array [
                        1,
                        0,
                        0,
                        1,
                        0,
                        0,
                      ]
                    }
                    opacity={1}
                    originX={0}
                    originY={0}
                    propList={
                      Array [
                        "fill",
                      ]
                    }
                    rotation={0}
                    scaleX={1}
                    scaleY={1}
                    skewX={0}
                    skewY={0}
                    stroke={null}
                    strokeDasharray={null}
                    strokeDashoffset={null}
                    strokeLinecap={0}
                    strokeLinejoin={0}
                    strokeMiterlimit={4}
                    strokeOpacity={1}
                    strokeWidth={1}
                    vectorEffect={0}
                    x={0}
                    y={0}
                  />
                </RNSVGGroup>
              </RNSVGSvgView>
              <View
                style={
                  Object {
                    "paddingHorizontal": 4,
                  }
                }
                withIcon="text"
              >
                <Text
                  style={
                    Object {
                      "color": "#00676D",
                      "fontFamily": "InterUI-Bold",
                      "fontSize": 12,
                      "lineHeight": 18,
                    }
                  }
                  tint="#00676D"
                >
                  title
                </Text>
              </View>
            </View>
            <View>
              <Text
                style={
                  Object {
                    "color": "#303030",
                    "fontFamily": "InterUI-Bold",
                    "fontSize": 16,
                    "lineHeight": 24,
                    "paddingBottom": 12,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "#303030",
                      "fontFamily": "InterUI-Bold",
                      "fontSize": 16,
                      "lineHeight": 24,
                    }
                  }
                >
                  John 3:16
                </Text>
                <Text
                  style={
                    Object {
                      "color": "#505050",
                      "fontFamily": "InterUI-Bold",
                      "fontSize": 12,
                      "lineHeight": 18,
                    }
                  }
                >
                   
                  WEB
                </Text>
              </Text>
              <View
                style={
                  Object {
                    "paddingBottom": 18,
                  }
                }
              >
                <View>
                  <View
                    style={
                      Object {
                        "paddingBottom": 0,
                      }
                    }
                  >
                    <Text
                      bold={false}
                      italic={false}
                      style={
                        Object {
                          "color": "#303030",
                          "fontFamily": "DroidSerif",
                          "fontSize": 16,
                          "lineHeight": 26,
                        }
                      }
                    >
                      <Text
                        bold={false}
                        italic={false}
                        style={
                          Object {
                            "color": "#505050",
                            "fontFamily": "DroidSerif",
                            "fontSize": 9.6,
                            "lineHeight": 26,
                          }
                        }
                      >
                         
                        <Text>
                          16
                        </Text>
                        Â 
                      </Text>
                      <Text
                        style={
                          Object {
                            "color": "#8b0000",
                          }
                        }
                      >
                        <Text>
                          For God so loved the world, that he gave his only born
                        </Text>
                      </Text>
                      <Text
                        bold={false}
                        italic={false}
                        style={
                          Object {
                            "color": "#303030",
                            "fontFamily": "DroidSerif",
                            "fontSize": 16,
                            "lineHeight": 26,
                          }
                        }
                      >
                        <Text>
                           
                        </Text>
                      </Text>
                      <Text
                        style={
                          Object {
                            "color": "#8b0000",
                          }
                        }
                      >
                        <Text>
                          Son, that whoever believes in him should not perish, but have eternal life, forever. 
                        </Text>
                      </Text>
                      <Text
                        bold={false}
                        italic={false}
                        style={
                          Object {
                            "color": "#303030",
                            "fontFamily": "DroidSerif",
                            "fontSize": 16,
                            "lineHeight": 26,
                          }
                        }
                      >
                        <Text>
                           
                        </Text>
                      </Text>
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "flexWrap": "wrap",
                "justifyContent": "flex-end",
                "paddingHorizontal": 16,
                "paddingTop": 0,
                "paddingVertical": 16,
              }
            }
          />
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
